sw/device/examples/sram_program/BUILD

```diff
@@ -42,23 +42,84 @@ opentitan_gdb_fpga_cw310_test(
     exit_success_pattern = "sram_program\\.c:47\\] PC: 0x1000208c, SRAM: \\[0x10000000, 0x10020000\\)",
     gdb_script = """
         target extended-remote :3333
-        set pagination off
 
         echo :::: Send OpenOCD the 'reset halt' command.\\n
         monitor reset halt
 
-        echo :::: Set RAM as RWX by executing code between _start and test_rom_main()\\n
-        file test_rom.elf
-        break rom_test_main
-        continue
-        delete breakpoints
+
+        # Prior to connecting OpenOCD and GDB, the device is just chugging
+        # along. It has almost certainly already run its startup routine and
+        # configured the watchdog timer. This is a problem; if the timer fires,
+        # the corresponding exception will steal control away from the debugger
+        # and cause this test to fail.  Thus, we need to disable the watchdog
+        # timer as quickly as possible to minimize the chances of it firing.
+        #
+        # The proper solution would be to disable ROM execution by provisioning
+        # with an OTP image that has CREATOR_SW_CFG_ROM_EXEC_EN set to zero.
+        #
+        # Initialize and disable the watchdog timers per aon_timer docs [1]. The
+        # hardcoded addresses were computed by adding REG_OFFSET values from
+        # //hw/ip/aon_timer/data:aon_timer_regs to the base address
+        # TOP_EARLGREY_AON_TIMER_AON_BASE_ADDR.
+        #
+        # [1]: https://docs.opentitan.org/hw/ip/aon_timer/doc/#initialization
+
+        # Disable WDOG_CTRL.
+        monitor mdw 0x40470014
+        monitor mww 0x40470014 0
+        # Disable WKUP_CTRL.
+        monitor mdw 0x40470004
+        monitor mww 0x40470004 0
```

Comment:
We don't need to worry about the wakeup timer since interrupts are disabled globally at reset.
```suggestion
```